### PR TITLE
feat: support virtual tables with expressions

### PR DESCRIPTION
**Pull request by srikrishnak**

### CODE REVIEW

File: plan/virtual_table_from_expr_test.go [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+		URI:  "https://github.com/substrait-io/substrait/blob/main/extensions/functions_arithmetic.yaml",
+		Name: "add:i32_i32",
+	}
+	b.GetFunctionRef(id.URI, id.Name)
+	scalarExpr, err := b.ScalarFn(id.URI, id.Name, nil, val1, val2)
+	require.NoError(t, err)
+	return scalarExpr
+}
+
+func buildLiteralExpressions(t *testing.T, b plan.Builder) []expr.VirtualTableExpressionValue {
```
Comment (anshuldata) [inline plan/virtual_table_from_expr_test.go]: nit: unused b so remove

File: plan/testdata/select_from_value_scalar.json [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+                  }
+                }
+              }
+            ]
+          }
+        },
+        "names": [
+          "col0",
+          "col1",
+          "col0",
```
Comment (anshuldata) [inline plan/testdata/select_from_value_scalar.json:140]: why colums are repeated ?

File: plan/testdata/select_from_value_literal.json [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+                  }
+                }
+              }
+            ]
+          }
+        },
+        "names": [
+          "col0",
+          "col1",
+          "col0",
```
Comment (anshuldata) [inline plan/testdata/select_from_value_literal.json:82]: why colums are repeated ?

File: plan/testdata/select_from_value_scalar.json (hunk 2 above)
Comment (srikrishnak) [reply plan/testdata/select_from_value_scalar.json:140]: This is as per susbtrait's project rel behaviour.

File: plan/testdata/select_from_value_literal.json (hunk 3 above)
Comment (srikrishnak) [reply plan/testdata/select_from_value_literal.json:82]: This is as per susbtrait's project rel behaviour.

**Pull request closed without merging**